src/App.jsx

```diff
@@ -14,45 +17,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+function resultGoodsArr(goods, { sortField, reverse }) {
+  const resultingArr = [...goods];
+
+  if (sortField) {
+    resultingArr.sort((good1, good2) => {
+      switch (sortField) {
+        case (SORT_FIELD_ALPHABET):
+          return good1.localeCompare(good2);
+        case (SORT_FIELD_LENGTH):
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    resultingArr.reverse();
+  }
+
+  return resultingArr;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  // eslint-disable-next-line
+  const goods = resultGoodsArr(goodsFromServer, { sortField, reverse, setReverse });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const reset = () => {
+    setSortField('');
+    setReverse(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info',
+            {
+              'is-light': sortField !== SORT_FIELD_ALPHABET,
+            })}
+          onClick={(() => setSortField(SORT_FIELD_ALPHABET))}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success',
+            {
+              'is-light': sortField !== SORT_FIELD_LENGTH,
+            })}
+          onClick={(() => setSortField(SORT_FIELD_LENGTH))}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning',
+            {
+              'is-light': reverse !== true,
+            })}
+          onClick={() => setReverse(!reverse)}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverse) && (
```

Comment:
you can create separate variable like `isShowResetButton`
```suggestion
        const isShowResetButton = sortField || isReverse;
...
        {isShowResetButton && (
```